### ISSUE TITLE
`key` validation incorrect when deletion - should not allow deleting attrs that are keys

### ISSUE THREAD

**Issue by flyingsilverfin** [vaticle/typedb#5255]
## Description
It is possible to work around `key` requirements (each schema element defined to have a key must have 1 of these keys) using `delete`. We can insert and then delete the attribute that is the key.

## Environment
Grakn Core 1.5.5

## Reproducible Steps
Define schema:
```
define person sub entity, key unique-key;
unique-key sub attribute, datatype long;
```
Then insert an instance:
`insert $x isa person, has unique-key 100;`

commit

Then delete the key:
`match $x isa unique-key; $x 100; delete $x;`
commit

This works.

## Expected Output
Validation error

## Actual Output
Successfully deletes the key `100`, leaving the person without a key.
 
## Additional information
This can get quite complicated, for instance what happens if two schema elements share the same key type.
eg:
```
define person sub entity, key unique-key;
house sub entity, key unique-key;
```
and 
`insert $x isa person, has unique-key 100; $y isa house, has unique-key 100;`
commit
then delete:
`match $x isa person, has unique key $y; delete $x, $y;`
We have to validate that the key instance is not being used for any other type as a key as well, before allowing the deletion to proceed!

Also consider for instance of one type uses the attribute as a key and the other as a normal attribute, one provides a deletion restriction but the other not.

**Comment by alexjpwalker** [comment]
This should be fixed in 2.0.

**Comment by krishnangovindraj** [comment]
This has since been fixed. Tested on 2.14.3:

```
> transaction issue_cleanup_scratch data write
issue_cleanup_scratch::data::write> match $x isa unique-key; $x 100; 
                                    delete $x isa unique-key;
                                    
Deleted from 1 matched answer(s)
issue_cleanup_scratch::data::write*> commit
[THW08] Invalid Thing Write: Attempted to commit a(n) 'person' that is missing key(s) of type(s): 'unique-key'
```
A test exists in the behaviour repository: `Scenario: deleting an attribute instance that is owned as a has throws @key an error`

Further,
```
insert
 $p1 isa person, has name "t";
 $p2 isa person, has name "t";

## Error> [THW07] Invalid Thing Write: Attempted to assign a key 't1' of type 'name' that had been taken by another 'person'.
```
